### PR TITLE
[GH-1278] Fix SG update-workload! implementation from updating Clio multiple times

### PR DESCRIPTION
**Pull request by ehigham**
[GH-1278] Fix SG update-workload! implementation from trying to update clio BAM records for each Succeeded workflow repeatedly by restoring the :finished guard for register-workload-in-clio.
Introducted in [GH-1277].

[GH-1278]: https://broadinstitute.atlassian.net/browse/GH-1278

[GH-1277]: https://broadinstitute.atlassian.net/browse/GH-1277

### CODE REVIEW

File: api/src/wfl/module/sg.clj [hunk 1]
```diff
@@ -191,21 +191,21 @@
 
 ;; visible for testing
 (defn register-workload-in-clio
-  "Use `tx` to register `_workload` outputs with Clio."
-  [{:keys [executor items output] :as _workload} tx]
-  (run!
-   (partial register-workflow-in-clio executor output)
-   (postgres/get-table tx items)))
+  [{:keys [executor output workflows] :as _workload}]
+  (run! (partial register-workflow-in-clio executor output) workflows))
```
Comment (tbl3rd) [inline api/src/wfl/module/sg.clj:195]: Restore the docstring or remove `:as _workload`.

File: api/src/wfl/module/sg.clj [hunk 2]
```diff
@@ -191,21 +191,21 @@
...
             (postgres/update-workload-status! tx workload)
-            (doto (workloads/load-workload-for-id tx id)
-              (register-workload-in-clio tx)))]
-    (if (and started (not finished)) (update! workload) workload)))
+            (workloads/load-workload-for-id tx id))]
+    (if (and started (not finished))
+      (let [workload' (update! workload)]
+        (when (:finished workload') (register-workload-in-clio workload'))
+        workload')
+      workload)))
```
Comment (tbl3rd) [inline api/src/wfl/module/sg.clj:208]: `workload'` is kinda weird, but I get it.

File: api/test/wfl/integration/modules/sg_test.clj [hunk 3]
```diff
@@ -382,11 +377,14 @@
...
-     #'sg/register-workload-in-clio             (fn [x _] x)}
-    #(shared/run-workload-state-transition-test! (the-sg-workload-request))))
+  (let [count           (atom 0)
+        increment-count (fn [_] (swap! count inc))]
+    (with-redefs-fn
+      {#'cromwell/submit-workflows                mock-cromwell-submit-workflows
+       #'postgres/batch-update-workflow-statuses! mock-batch-update-workflow-statuses!
+       #'sg/register-workload-in-clio             increment-count}
+      #(shared/run-workload-state-transition-test! (the-sg-workload-request)))
+    (is (= 1 @count) "Clio was updated more than once")))
```
Comment (tbl3rd) [inline api/test/wfl/integration/modules/sg_test.clj:387]: Could use `==` here.

**Review by tbl3rd — APPROVED** [review]
OK.  Will collapsing the 2 updates make your db/IO refactoring slightly more challenging?

File: api/src/wfl/module/sg.clj (hunk 1 above)
Comment (ehigham) [reply api/src/wfl/module/sg.clj:195]: I don't understand when you want a `_workload`? I thought it was for more than just the docstring.

**Comment by ehigham** [comment]
> OK. Will collapsing the 2 updates make your db/IO refactoring slightly more challenging?

Thanks for taking a look. No - I don't think so.

File: api/src/wfl/module/sg.clj (hunk 1 above)
Comment (tbl3rd) [reply api/src/wfl/module/sg.clj:196]: I don't always understand either.  Sometimes I include it to clarify what the keywords belong to.  Sometimes it's to support the docstring.  Sometimes it's both.

Comment (tbl3rd) [reply api/src/wfl/module/sg.clj:196]: BTW: Thank you for restoring the docstring.

**Pull request merged**